Loadbalancer changes

ramaraosrikakulapu opened the pull request:
Following are the changes made - 

- Updated load balancer configurations
- Added use case for load balancer

ayasuda2OO3 commented on spec/loadbalancer/default.conf:
```diff
@@ -1,27 +1,22 @@
 upstream master {
-     server 192.168.1.116:17990/agent;
-     server 192.168.1.105:17990/agent;
+     server IP1:17990;
```
Are the IP1/IP2  the env variables to the respective IP in the machine.env file? @ramaraosrikakulapu

ayasuda2OO3 replied:
1) To use environment vars, it should be templitised as specified here https://serverfault.com/a/755541 This may break the image build. 
2) Please provide the **test build log**, if this believed to be a working setting.

ramaraosrikakulapu replied:
> Are the IP1/IP2 the env variables to the respective IP in the machine.env file? @ramaraosrikakulapu

@ayasuda2003 Thanks for review. Here are my comments - 
- Yes, IP1/IP2 are supposed to read from machine.env file
- Templatizing the conf file will taken care in future MR's. This MR includes, adding path based configuration and documentation

ayasuda2OO3 replied:
Understand. Please propose a PR along with a test build log when the issue is resolved. Meanwhile, this file is not usable in this container spec.

ramaraosrikakulapu replied:
Sure chia.. Will keep open the PR and will continue to take the suggestions

ramaraosrikakulapu replied:
default.conf is now templatized in latest commit. We just have to update IP addresses in in machine.env and pass as env-file to docker run command.

ayasuda2OO3 commented on spec/loadbalancer/README.md:
````diff
@@ -23,9 +23,52 @@ docker run -p 8080:80 --env-file $(pwd)/machine.env enterpriseconnect/loadbalanc
 ```
 
````
Please provide the following-
1) The use case Spec/requirement.
2) Test script, steps, etc.

@ramaraosrikakulapu @palokam

ayasuda2OO3 replied:
> > The diagram lb-usecase.png does not depict the solution in detail. please purpose a clearer diagram @palokam @ramaraosrikakulapu
> 
> Can you please be more specific?

Sure. in the proposed diagram, 1) what type of traffic to be load-balanced, 2) what is the EC Server/Client usage/environment? 3) what is Nginx to do with EC LoadBalancer connectivity? 4) what does the word "EC" in the cloud try to symbolise? etc.

ramaraosrikakulapu replied:
> > > The diagram lb-usecase.png does not depict the solution in detail. please purpose a clearer diagram @palokam @ramaraosrikakulapu
> > 
> > 
> > Can you please be more specific?
> 
> Sure. in the proposed diagram, 1) what type of traffic to be load-balanced, 2) what is the EC Server/Client usage/environment? 3) what is Nginx to do with EC LoadBalancer connectivity? 4) what does the word "EC" in the cloud try to symbolise? etc.

Updated with new diagram as discussed for Phase 1. I will update the diagram once we ready with Phase 2.

ayasuda2OO3 replied:
> 3) what is Nginx to do with EC LoadBalancer connectivity

Please replace the diagram ```doc/lb-usecase.png``` with ```doc/lb-model.png```, or make improvement based on the ```doc/lb-model.png``` (find in lucidchart)  the proposed diagram does not seem appeal nor indifference. Thanks! @ramaraosrikakulapu

ramaraosrikakulapu replied:
Agree. We are trying to implement the same as described in ```docs/lb-model.png```. So I preferred not to have diagram for use case.

ayasuda2OO3 replied:
can you move the figure ```doc/lb-model.png``` to this use case section instead?

ramaraosrikakulapu commented:
> The diagram lb-usecase.png does not depict the solution in detail. please purpose a clearer diagram @palokam @ramaraosrikakulapu

Can you please be more specific?

ayasuda2OO3 commented:
> > The diagram lb-usecase.png does not depict the solution in detail. please purpose a clearer diagram @palokam @ramaraosrikakulapu
> 
> Can you please be more specific?
### Re-post
Sure. in the proposed diagram, 1) what type of traffic to be load-balanced, 2) what is the EC Server/Client usage/environment? 3) what is Nginx to do with EC LoadBalancer connectivity? 4) what does the word "EC" in the cloud try to symbolise? etc.

ramaraosrikakulapu commented:
> could you provide the build log `docker build ...` for the image building?

Here are the docker build logs - 
[buildlogs.txt](https://github.com/Enterprise-connect/oci/files/4587581/buildlogs.txt)

ayasuda2OO3 commented:
> > could you provide the build log `docker build ...` for the image building?
> 
> Here are the docker build logs -
> [buildlogs.txt](https://github.com/Enterprise-connect/oci/files/4587581/buildlogs.txt)

Looking good. Could you also put up an explaination for the available tag(s) in the readme? E.g. https://github.com/Enterprise-connect/oci#agent-in-containers

ramaraosrikakulapu commented:
> > > could you provide the build log `docker build ...` for the image building?
> > 
> > 
> > Here are the docker build logs -
> > [buildlogs.txt](https://github.com/Enterprise-connect/oci/files/4587581/buildlogs.txt)
> 
> Looking good. Could you also put up an explaination for the available tag(s) in the readme? E.g. https://github.com/Enterprise-connect/oci#agent-in-containers

Added the tags information as suggested. Also updated the documentation with more details.

ayasuda2OO3 commented on spec/loadbalancer/README.md:
````diff
@@ -22,11 +20,65 @@ enterpriseconnect/loadbalancer:v1.1beta
+- [```v1.1beta```](https://github.com/Enterprise-connect/oci/blob/v1.1beta/spec/loadbalancer/Dockerfile)
+
+#### tag usage
+- ```v1.1beta``` refers to the image to build agent ```#2730+```-relate releases.
````
What does it to do with the agent and why? @ramaraosrikakulapu

ramaraosrikakulapu replied:
To run watcher, we have to run command ```./agent -cfg gw.yml -wtr``` which requires agent to download

ayasuda2OO3 replied:
consider this- 

```v1.1beta``` is a beta release which is compatible with and only created for the EC agent in beta usage. It does not load balance traffic from any EC **production** releases. E.g. ```v1```, ```v1.1```, ```v1.2```, etc.

ayasuda2OO3 replied:
The ```tag usage``` is still missing in the latest commit. @ramaraosrikakulapu

ayasuda2OO3 commented on spec/loadbalancer/README.md:
```diff
@@ -1,4 +1,4 @@
-## EC Load Balancer Spec
+## EC Gateway Load Balancer Spec
 ### Design
```
@ramaraosrikakulapu Please add **Feature**, and **Goal** section as below example- 

## Goal
E.g. Load balancing traffic for EC connectivity as service.

## Feature
Cloud platform-agnostic, etc.

ayasuda2OO3 replied:
@ramaraosrikakulapu proposed ```Goal``` looks good. Still need content for the ```Feature``` . And please help add ```Runtime Requirement``` to the readme.

ayasuda2OO3 commented:
@ramaraosrikakulapu As for the last step, please help configure the CI for building out the image https://hub.docker.com/repository/docker/enterpriseconnect/loadbalancer/builds

ayasuda2OO3 commented on spec/loadbalancer/README.md:
```diff
@@ -1,14 +1,10 @@
-## EC Load Balancer Spec
+## EC Gateway Load Balancer Spec
```
@ramaraosrikakulapu please adding docker build badge on the top of the readme as follows-

```![Docker Cloud Build Status](https://img.shields.io/docker/cloud/build/enterpriseconnect/loadbalanceri)```

ramaraosrikakulapu replied:
Docker build status added

ayasuda2OO3 commented on spec/loadbalancer/README.md:
```diff
@@ -1,14 +1,17 @@
+### Runtime Requirement
+- VM's with same network group/ subnet. One VM for load balancer and rest for watchers
+- OS: Linux, Windows, Darwin, Arm, etc.
+- [docker 1x.xx+](https://docs.docker.com/get-docker/)   
```
Please consider remove-
> VM's with same network group/ subnet. One VM for load balancer and rest for watchers

```docker 1x.xx+ ``` is not a valid version. Please verify in official docker versions.

ramaraosrikakulapu replied:
Updated the version

ayasuda2OO3 commented on spec/loadbalancer/README.md:
```diff
@@ -1,14 +1,17 @@
+- VM's with same network group/ subnet. One VM for load balancer and rest for watchers
+- OS: Linux, Windows, Darwin, Arm, etc.
+- [docker 1x.xx+](https://docs.docker.com/get-docker/)   
+- [EC Agent](#tag-usage)
```
what is tag usage? where is the compatibility? v1.1 vs v1.1beta, etc.

ramaraosrikakulapu replied:
Added compatability details in tabular format

ayasuda2OO3 commented on spec/loadbalancer/README.md:
```diff
@@ -22,11 +25,74 @@ enterpriseconnect/loadbalancer:v1.1beta
+
 ### Use-case I vm-2-vm
-vm to vm load-balancing for agent file trasfer.
+![LB High Level](/doc/lb-model.png)
```
Please give it a high-level introduction of what this use-case about @ramaraosrikakulapu

ramaraosrikakulapu replied:
Added high level focus of use case
